### PR TITLE
feat(custom-elements): include build utils

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

This includes utilities for consumers of the [`dist-custom-elements-bundle`](https://stenciljs.com/docs/custom-elements) output target, namely for:

1. Registering individual custom elements with their respective tag
2. Setting the asset path (this is a wrapper for Stencil's [`setAssetPath`](https://stenciljs.com/docs/custom-elements#making-assets-available), so we could drop if deemed redundant/not useful)
3. Entry-point modules that import and register the root and supporting components. This will make it easier for devs to include a single bundle and not have to think/maintain a list of supporting component dependencies.

These are currently placed under `dist/custom-elements/bundles/`

Also, this adds a JSON output of the doc, which includes metadata for each of the components under an `extras` folder (`dist/extras`). This could be useful for other devs if they wanted to build tooling for our components.

### Pending

I will tackle the following after the first round of feedback.

- [x] wire up util script with main build
- [x] tidy up bundle generator
- [ ] ~minify output~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "util:prep-next-from-existing-build": "ts-node ./support/prepareVersionUpdate.ts --next --standard-version-options=\"--skip.changelog\"",
     "util:publish-next": "npm publish --tag next",
     "util:push-tags": "git push --follow-tags",
+    "util:build-custom-elements-output-utils": "ts-node support/output-targets/custom-elements/createWrappers.ts && tsc --project ./support/output-targets/custom-elements/tsconfig-dist-utils.json && rimraf ./extras/output-targets/custom-elements/wrappers",
```
Comment (paulcpederson) [inline package.json:52]: may be worth breaking this into multiple commands or just multiple lines? Very long.

**Review by paulcpederson — APPROVED** [review]

**Comment by paulcpederson** [comment]
Nice this will be really useful.

**Review by driskull — COMMENTED** [review]
This looks good to me.

File: package.json (hunk 1 above)
Comment (jcfranco) [reply package.json:52]: Agreed, some of these are getting unwieldy. 

I'll tackle this in a follow-up PR. https://www.npmjs.com/package/script-launcher could help break some of these into multiple lines, which would improve readability.

**Pull request merged**